compiler/optimizer/VPHandlers.cpp

```diff
@@ -1516,9 +1525,7 @@ TR::Node *constrainAload(OMR::ValuePropagation *vp, TR::Node *node)
       return node;
 
    // before undertaking aload specific handling see if the default load transformation helps
-   // But even if it does help, node may still be an aload that we can constrain.
-   if (TR::TransformUtil::transformDirectLoad(vp->comp(), node)
-       && node->getOpCodeValue() != TR::aload)
+   if (TR::TransformUtil::transformDirectLoad(vp->comp(), node))
```

Comment:
Why did we drop the != TR::aload test ?

Reply:
If `transformDirectLoad` has done something, it must have changed the node to a constant. For non-collected reference, it'll be an `aconst` node; for collected reference, it'll be a known object index on the symref of the `TR::aload` node. Both cases can be handled by `constrainNewlyFoldedConst` now. Notice this commit contains a change to `constrainNewlyFoldedConst` to create known object constraint.

Reply:
Okay so it is specifically to allow the known object constraint to be created in the "current" pass then.